src/packagedcode/pypi.py

```diff
@@ -187,11 +189,25 @@ def parse_dependencies(location, package):
             package.dependencies = dependencies
 
 
+def file_type(file_name):
```

Comment:
What about this may be? which will also support adding other dparse parsers?
```
dependency_type_by_extensions = {
    ('.txt', '.in'): 'requirements.txt',
}

def get_dependency_type(file_name, dependency_type_by_extensions=dependency_type_by_extensions):
    """
    Return the type of a dependency as a string or None given a `file_name` string.
    """
    for extensions, dependency_type in dependency_type_by_extensions.items():
        if file_name.endswith(extensions):
            return dependency_type
```

Reply:
I was also about to change this like this. This will be more easier.
I will change this shortly :)